src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1771,6 +1780,45 @@ bool OutputStateMachineEngine::_GetOscSetColorTable(const std::wstring_view stri
     return success;
 }
 
+// Routine Description:
+// - Given a hyperlink string, attempts to parse the URI encoded. An 'id' parameter
+//   may be provided.
+//   If there is a URI, the well formatted string looks like:
+//          "<params>;<URI>"
+//   If there is no URI, we need to close the hyperlink and the string looks like:
+//          ";"
+// Arguments:
+// - string - the string containing the parameters and URI
+// - params - where to store the parameters
+// - uri - where to store the uri
+// Return Value:
+// - True if a URI was successfully parsed or if we are meant to close a hyperlink
+bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
+                                               std::wstring& params,
+                                               std::wstring& uri) const
+{
+    params.clear();
+    uri.clear();
+    const std::wstring idStr = L"id=";
+    const auto len = string.size();
+    const size_t midPos = string.find(';');
+    if (midPos != std::wstring::npos)
+    {
+        if (len != 1)
+        {
+            uri = string.substr(midPos + 1);
+            const auto paramStr = string.substr(0, midPos);
+            const auto idPos = paramStr.find(idStr);
+            if (idPos != std::wstring::npos)
+            {
+                params = paramStr.substr(idPos + idStr.size());
+            }
+        }

```

Comment:
So, is this getting just the text after `id=` as the value of `param`? What if the string is something like `id=foo:bar=baz;www...`? 

It doesn't look like we're using the param string anywhere yet so it's probably nbd, but just a question.

Reply:
According to the hyperlink [gist](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda) (which I've been taking as the spec), currently only the `id` parameter is defined - so yep we just take that and ignore any others.